lib/file-helper.js

```diff
@@ -9,75 +9,148 @@ const configPaths = require('../config/paths.js')
+ *
+ * @param {string} directoryPath
+ * @returns {Promise<Map<dirname, Map<filename, { path: string; stats: import('fs').Stats }>>>} File entries by directory
+ */
+const getFilesByDirectory = async (directoryPath) => {
+  const directories = await getDirectories(directoryPath)
+
+  // Directory listing per directory
+  const listings = await Promise.all([...directories]
+    .map(async ([basename, { path }]) => [basename, await getFiles(path)])
+  )
+
+  return new Map(listings)
+}
 
-const getComponentData = componentName => {
+/**
+ * Load single component data
+ *
+ * @param {string} componentName - Component name
+ * @returns {Promise<ComponentData>} Component data
+ */
+const getComponent = async (componentName) => {
   try {
-    const yamlPath = path.join(configPaths.components, componentName, `${componentName}.yaml`)
-    return yaml.load(
-      readFileSync(yamlPath, 'utf8'), { json: true }
-    )
+    const yamlPath = join(configPaths.components, componentName, `${componentName}.yaml`)
+    const componentData = yaml.load(await readFile(yamlPath, 'utf8'), { json: true })
+
+    return {
+      name: componentName,
+      ...componentData
+    }
   } catch (error) {
     throw new Error(error)
   }
 }
 
-const fullPageExamples = () => {
-  return getDirectories(path.resolve(configPaths.fullPageExamples))
-    .map(folderName => ({
-      name: folderName,
-      path: folderName,
-      ...fm(readFileSync(path.join(configPaths.fullPageExamples, folderName, 'index.njk'), 'utf8')).attributes
-    }))
-    .sort((a, b) => (a.name.toLowerCase() > b.name.toLowerCase()) ? 1 : -1)
+/**
+ * Load all components' data
+ *
+ * @returns {Promise<ComponentData>[]} Components' data
+ */
+const getComponents = async () => {
+  const directories = await getDirectories(configPaths.components)
+  return Promise.all([...directories.keys()].map(getComponent))
+}
+
+/**
+ * Load all full page examples' front matter
+ *
+ * @returns {Promise<{ name: string; scenario?: string; notes?: string }[]>} Full page examples
+ */
+const getFullPageExamples = async () => {
+  const directories = await getDirectories(configPaths.fullPageExamples)
+
+  // Add metadata (front matter) to each example
+  const examples = await Promise.all([...directories.keys()].map(async (exampleName) => {
+    const templatePath = join(configPaths.fullPageExamples, exampleName, 'index.njk')
+    const { attributes } = fm(await readFile(templatePath, 'utf8'))
+
+    return {
+      name: exampleName,
+      ...attributes
+    }
+  }))
+
+  const collator = new Intl.Collator('en', {
+    sensitivity: 'base'
+  })
+
+  return examples.sort(({ name: a }, { name: b }) =>
+    collator.compare(a, b))
 }
 
 module.exports = {
-  allComponents,
-  fullPageExamples,
-  getComponentData,
+  getComponent,
+  getComponents,
   getDirectories,
   getFiles,
+  getFilesByDirectory,
+  getFullPageExamples,
   getListing
 }
+
+/**
+ * Component data from YAML
+ *
+ * @typedef {object} ComponentData
```

Comment:
For editors and IDEs that support JSDoc this is the bit where object types can be discovered

```mjs
const component = await getComponent(componentName)

component.name
component.params
component.previewLayout
component.accessibilityCriteria
```

<img width="360" alt="JSDoc @typedef showing all available component properties" src="https://user-images.githubusercontent.com/415517/194050382-f9805c5d-14f8-4bc3-8b2e-354343bed816.png">